src/solvers/fd_tree/fd_2d.jl

```diff
@@ -0,0 +1,199 @@
+
+# General interface methods for SummationByPartsOperators.jl and Trixi.jl
+nnodes(D::AbstractDerivativeOperator) = size(D, 1)
+eachnode(D::AbstractDerivativeOperator) = Base.OneTo(nnodes(D))
+get_nodes(D::AbstractDerivativeOperator) = grid(D)
+
+
+# For dispatch
+const DGFD = DG{Basis} where {Basis<:AbstractDerivativeOperator}
+
+
+# 2D containers
+init_mortars(cell_ids, mesh, elements, mortar::Nothing) = nothing
+
+
+# 2D caches
+function create_cache(mesh::TreeMesh{2}, equations,
+                      volume_integral::VolumeIntegralStrongForm, dg, uEltype)
+
+  prototype = Array{SVector{nvariables(equations), uEltype}, ndims(mesh)}(
+    undef, ntuple(_ -> nnodes(dg), ndims(mesh))...)
+  f_threaded = [similar(prototype) for _ in 1:Threads.nthreads()]
+
+  return (; f_threaded,)
+end
+
+create_cache(mesh, equations, mortar::Nothing, uEltype) = NamedTuple()
+nmortars(mortar::Nothing) = 0
+
+
+# 2D RHS
+function calc_volume_integral!(du, u,
+                               mesh::TreeMesh{2},
+                               nonconservative_terms::Val{false}, equations,
+                               volume_integral::VolumeIntegralStrongForm,
+                               dg::DGFD, cache)
+  D = dg.basis # SBP derivative operator
+  @unpack f_threaded = cache
+  if nvariables(equations) == 1
+    # `reinterpret(reshape, ...)` removes the leading dimension only if more
+    # than one variable is used.
+    u_vectors  = reshape(reinterpret(SVector{nvariables(equations), eltype(u)}, u),
+                         nnodes(dg), nnodes(dg), nelements(dg, cache))
+    du_vectors = reshape(reinterpret(SVector{nvariables(equations), eltype(du)}, du),
+                         nnodes(dg), nnodes(dg), nelements(dg, cache))
+  else
+    u_vectors  = reinterpret(reshape, SVector{nvariables(equations), eltype(u)}, u)
+    du_vectors = reinterpret(reshape, SVector{nvariables(equations), eltype(du)}, du)
+  end
+
+  @threaded for element in eachelement(dg, cache)
+    f_element = f_threaded[Threads.threadid()]
+    u_element = view(u_vectors,  :, :, element)
+
+    # x direction
```

Comment:
Maybe add a high-level comment explaining the numerical operation here? Something like
```suggestion
    # Line-by-line calculation of `flux × D + dU`
    # x direction
```
but prettier?

Reply:
Tack!